packages/axelar-wasm-std/src/voting.rs

```diff
@@ -275,7 +284,7 @@ impl Poll for WeightedPoll {
         // late votes are not tallied
         if self.status != PollStatus::InProgress {
             participation.vote = Some(votes);
-            return Ok(());
+            return Ok(self);
         }
```

Comment:
do we actually need this?
Basically, what we decided was that the poll is open until it expires, you can just query for a result early

Reply:
I didn't wanted to mix refactoring PR with logic that changes behavior, this will be dealt in another PR. I just added a TODO meanwhile